### ISSUE TITLE
Error when loading Resnet_v1_50.ckpt in multiple GPUs machine

### ISSUE THREAD

**Issue by Nikasa1889** [CharlesShang/FastMaskRCNN#36]
While running the train/train.py code on a 4-GPUs machine, I have got the following error:

`Restored 267(544) vars from ./data/pretrained_models/resnet_v1_50.ckpt
2017-04-27 15:38:28.434278: W tensorflow/core/framework/op_kernel.cc:1152] Invalid argument: Input to reshape is a tensor with 2186240 values, but the requested shape has 546560
         [[Node: Reshape_1 = Reshape[T=DT_INT32, Tshape=DT_INT32, _device="/job:localhost/replica:0/task:0/gpu:0"](Cast_4/_615, Reshape_1/shape)]]
...
`
This must be an issue with muti-GPUs setting, since the number of values required is equal 1/4 of the provided values.

**Comment by Nikasa1889** [comment]
I have tried to mask the visible GPUs using this command:

`CUDA_VISIBLE_DEVICES=1 python train/train.py`

I still got the same error.

**Comment by kevinkit** [comment]
Can you see with `watch nvidia-smi` if he really masks the other GPUs or if there is still memory occupied on all GPUs by them?

**Comment by Nikasa1889** [comment]
The training output said that only gpu:0 was loaded. I will double check that tomorrow. I do not have this issue on single-GPU machine, which further confirms the multi-CPUs issue.

**Comment by CharlesShang** [comment]
Welcome multiple GPUs contributions!!!!!!
I dont have mult-gpu machines.. I guess it takes some efforts to get done. 
One example is [inceptionv3](https://github.com/tensorflow/models/tree/master/inception)

**Comment by CharlesShang** [comment]
[inceptionv3](https://github.com/tensorflow/models/tree/master/inception) uses a center device to store parameters, and creates one instance for each gpu (sharing same params). Then the params are updated using averaged gradients.

**Comment by Nikasa1889** [comment]
@CharlesShang Sorry, the multi-gpus server has been occupied by another project. I will probably have it back next week and will try to see if I can fix the problem following your suggestions.

**Comment by kevinkit** [comment]
Also see #17

**Comment by tjussh** [comment]
@Nikasa1889 hi, I have the similar error. Have you solved it yet?
